spec/index.bs

```diff
@@ -2785,18 +2785,50 @@ interface GPUTexture {
     GPUTextureView createView(optional GPUTextureViewDescriptor descriptor = {});
 
     undefined destroy();
+
+    [SameObject] readonly attribute GPUExtent3DDict size;
+    readonly attribute GPUIntegerCoordinate mipLevelCount;
+    readonly attribute GPUSize32 sampleCount;
+    readonly attribute GPUTextureDimension dimension;
+    readonly attribute GPUTextureFormat format;
+    readonly attribute GPUTextureUsageFlags usage;
 };
 GPUTexture includes GPUObjectBase;
 </script>
 
 {{GPUTexture}} has the following internal slots:
 
 <dl dfn-type=attribute dfn-for="GPUTexture">
-    : <dfn>\[[descriptor]]</dfn>, of type {{GPUTextureDescriptor}}
+    : <dfn>size</dfn>, of type {{GPUExtent3DDict}}
+    ::
+        The width, height, and depth or layer count of this {{GPUTexture}}.
+
+        Note: It is a {{GPUExtent3DDict}} with all fields set instead of a {{GPUExtent3D}} union type.
+
+    : <dfn>mipLevelCount</dfn>, of type {{GPUIntegerCoordinate}}
     ::
-        The {{GPUTextureDescriptor}} describing this texture.
+        The number of mip levels of this {{GPUTexture}}.
 
-        All optional fields of {{GPUTextureDescriptor}} are defined.
+    : <dfn>sampleCount</dfn>, of type {{GPUSize32}}
+    ::
+        The number of sample count of this {{GPUTexture}}.
+
+    : <dfn>dimension</dfn>, of type {{GPUTextureDimension}}
+    ::
+        The dimension of the set of texel for each of this {{GPUTexture}}'s subresources.
+
+    : <dfn>format</dfn>, of type {{GPUTextureFormat}}
+    ::
+        The format of this {{GPUTexture}}.
+
+    : <dfn>usage</dfn>, of type {{GPUTextureUsage}}
+    ::
+        The allowed usages for this {{GPUTexture}}.
+
+    : <dfn>\[[viewFormats]]</dfn>, of type sequence&lt;{{GPUTextureFormat}}&gt;
```

Comment:
Why is `viewFormats` the only descriptor member not exposed here?

Reply:
On the opposite side, I thought we weren't going expose format/usage either, only sizes and maybe dimension (["shape"](https://github.com/gpuweb/gpuweb/issues/1498#issuecomment-1109900079)). I'm fine with whatever, though I think that would make the distinction a little clearer between what we do and don't reflect.

Reply:
I didn't expose it because it seems not as useful a part of the shape of the texture. Also it would need canonicalization which would be annoying to write in the spec :)

The goal of the reflection for container objects is that if they are passed around, you are given the information that's necessary to act on them. Brandon's mipmap computation helper function shows that the usage is part of that information. And format too because you need to create render pipelines for that format.

That's also why texture views didn't make the list: they are used to restrict the view to a specific range of subresources, so the code you give it to should be agnostic to exactly which range you gave. (what does it matter that you pass the full view of a 3mip 1 layer 2d texture, or a 3 mip 1 layer view of a much larger texture?)